receiver/scraper_test.go

```diff
@@ -0,0 +1,183 @@
+		ProcessRuntimeMemstatsHeapReleased:  metricEnabled,
+		ProcessRuntimeMemstatsHeapSys:       metricEnabled,
+		ProcessRuntimeMemstatsLastPause:     metricEnabled,
+		ProcessRuntimeMemstatsLookups:       metricEnabled,
+		ProcessRuntimeMemstatsMallocs:       metricEnabled,
+		ProcessRuntimeMemstatsMcacheInuse:   metricEnabled,
+		ProcessRuntimeMemstatsMcacheSys:     metricEnabled,
+		ProcessRuntimeMemstatsMspanInuse:    metricEnabled,
+		ProcessRuntimeMemstatsMspanSys:      metricEnabled,
+		ProcessRuntimeMemstatsNextGc:        metricEnabled,
+		ProcessRuntimeMemstatsNumForcedGc:   metricEnabled,
+		ProcessRuntimeMemstatsNumGc:         metricEnabled,
+		ProcessRuntimeMemstatsOtherSys:      metricEnabled,
+		ProcessRuntimeMemstatsPauseTotal:    metricEnabled,
+		ProcessRuntimeMemstatsStackInuse:    metricEnabled,
+		ProcessRuntimeMemstatsStackSys:      metricEnabled,
+		ProcessRuntimeMemstatsSys:           metricEnabled,
+		ProcessRuntimeMemstatsTotalAlloc:    metricEnabled,
+	}
+	allMetricsDisabled = metadata.MetricsSettings{
+		ProcessRuntimeMemstatsBuckHashSys:   metricDisabled,
+		ProcessRuntimeMemstatsFrees:         metricDisabled,
+		ProcessRuntimeMemstatsGcCPUFraction: metricDisabled,
+		ProcessRuntimeMemstatsGcSys:         metricDisabled,
+		ProcessRuntimeMemstatsHeapAlloc:     metricDisabled,
+		ProcessRuntimeMemstatsHeapIdle:      metricDisabled,
+		ProcessRuntimeMemstatsHeapInuse:     metricDisabled,
+		ProcessRuntimeMemstatsHeapObjects:   metricDisabled,
+		ProcessRuntimeMemstatsHeapReleased:  metricDisabled,
+		ProcessRuntimeMemstatsHeapSys:       metricDisabled,
+		ProcessRuntimeMemstatsLastPause:     metricDisabled,
+		ProcessRuntimeMemstatsLookups:       metricDisabled,
+		ProcessRuntimeMemstatsMallocs:       metricDisabled,
+		ProcessRuntimeMemstatsMcacheInuse:   metricDisabled,
+		ProcessRuntimeMemstatsMcacheSys:     metricDisabled,
+		ProcessRuntimeMemstatsMspanInuse:    metricDisabled,
+		ProcessRuntimeMemstatsMspanSys:      metricDisabled,
+		ProcessRuntimeMemstatsNextGc:        metricDisabled,
+		ProcessRuntimeMemstatsNumForcedGc:   metricDisabled,
+		ProcessRuntimeMemstatsNumGc:         metricDisabled,
+		ProcessRuntimeMemstatsOtherSys:      metricDisabled,
+		ProcessRuntimeMemstatsPauseTotal:    metricDisabled,
+		ProcessRuntimeMemstatsStackInuse:    metricDisabled,
+		ProcessRuntimeMemstatsStackSys:      metricDisabled,
+		ProcessRuntimeMemstatsSys:           metricDisabled,
+		ProcessRuntimeMemstatsTotalAlloc:    metricDisabled,
+	}
+)
+
+func newMockServer(t *testing.T, responseBodyFile string) *httptest.Server {
+	file, err := os.Open(responseBodyFile)
+	require.NoError(t, err)
+	fileContents, err := ioutil.ReadAll(file)
+	require.NoError(t, err)
+	return httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		if req.URL.Path == "/debug/vars" {
+			rw.WriteHeader(200)
+			_, err := rw.Write(fileContents)
+			require.NoError(t, err)
+			return
+		}
+		rw.WriteHeader(404)
```

Comment:
I would suggest using `http.StatusNotFound `

Reply:
Done, also similarly for the 200 above.